fix(a11y): fixing MultiSelect no focus indicator

rjhenriquez opened the pull request:
Closes #6855

Fixing MultiSelect - accessibility issue - no focus indicator visible on clear button

#### Changelog

**Changed**

MultiSelect component

#### Testing / Reviewing

- Go to http://localhost:9000/?path=/story/multiselect--default
- Select a few elements
- Focus on the button to "Clear Selection" by keyboard, it should show an outline

Commit:
fix(a11y): fixing MultiSelect no focus indicator visible on clear button

joshblack reviewed (commented):
Actually, taking a look at the history for `_tag.scss` will this create a regression of: https://github.com/carbon-design-system/carbon/pull/4792 potentially?

tw15egan commented:
@joshblack it seems like that regression already exists. Perhaps we need to add a `pointer-events: none` to the disabled tag?

![Sep-28-2020 15-22-13](https://user-images.githubusercontent.com/11928039/94492174-856e7000-019e-11eb-8e9b-37ad3462cdee.gif)

aledavila reviewed (changes requested):
Ah now I see it. @rjhenriquez did you want to look at the possible fix TJ suggested ?

rjhenriquez committed:
fix(a11y): fixing disabled tag pointer events

rjhenriquez commented:
@aledavila @tw15egan `pointer-events: none;` added

tw15egan commented:
I'm not seeing the disabled focus anymore, but it seems like the rest of the changes are gone now? Not able to tab to the clear icon

joshblack commented:
bump @rjhenriquez when you get a sec to respond to above 👀

rjhenriquez committed:
fix(a11y): fixing MultiSelect no focus indicator visible v2

tw15egan commented:
Just a few small issues I'm still seeing:

- [x] Filter tag no longer receives `not-allowed` cursor styles when disabled. We may need to handle the onClick behavior in the React code and just return if disabled is true




- [x] Multiselect badge does not receive any disabled styles when the multiselect is disabled (existing bug, can be tackled in a separate PR)
<img width="393" alt="Screen Shot 2020-10-28 at 11 28 27 AM" src="https://user-images.githubusercontent.com/11928039/97480778-1f961500-1911-11eb-9af5-f8bdca8f37d7.png">

- [x] Multiselect badge loses focus styles when hovered over 

![multiselect](https://user-images.githubusercontent.com/11928039/97480745-16a54380-1911-11eb-8755-1074ff624ef1.gif)

rjhenriquez commented:
@tw15egan 

- Sadly, `pointer-events: none` will override `cursor: not-allowed`, making the cursor default to the normal one. Do we create another issue to handle the onClick behavior in the React code ?

- > Multiselect badge does not receive any disabled styles when the multiselect is disabled
I think it's now fixed

- > Multiselect badge loses focus styles when hovered over
The `.bx--tag--filter:hover` has no outline on hover, so I added a hover inside the focus state.. weird but it works.

rjhenriquez committed:
fix(a11y): fixing MultiSelect no focus indicator visible v3

tw15egan commented on packages/components/src/components/list-box/_list-box.scss:
```diff
@@ -498,13 +502,15 @@ $list-box-menu-width: rem(300px);
-  .#{$prefix}--list-box--disabled
-    .#{$prefix}--list-box__selection--multi
+  .#{$prefix}--list-box--disabled .#{$prefix}--list-box__selection--multi {
+    @include tag-theme($disabled-02, $disabled-01);
```
Seems like `tag-theme` is not imported in `_list-box.scss` and is causing some build errors. 

We'll need to add a `@import '../tag/mixins';` to the top of `_list-box.scss`

tw15egan commented:
@rjhenriquez I've gone ahead and updated this PR so we can get it out before the release tomorrow 👍

tw15egan committed:
fix(mulitselect): add tag mixin import

Commit:
Merge branch 'master' into 6855-multiselect-focus-fix

emyarod reviewed (approved):
thanks for the updates @tw15egan, looks good to me

emyarod committed:
Merge branch 'master' into 6855-multiselect-focus-fix
